client/web/web.go

```diff
@@ -307,31 +307,58 @@ func (s *Server) requireTailscaleIP(w http.ResponseWriter, r *http.Request) (han
 		return false
 	}
 
-	st, err := s.lc.StatusWithoutPeers(r.Context())
+	st, err := s.lc.Status(r.Context())
 	if err != nil {
 		s.logf("error getting status: %v", err)
 		http.Error(w, "internal error", http.StatusInternalServerError)
 		return true
 	}
 
-	var ipv4 string // store the first IPv4 address we see for redirect later
-	for _, ip := range st.Self.TailscaleIPs {
-		if ip.Is4() {
-			if r.Host == fmt.Sprintf("%s:%d", ip, ListenPort) {
-				return false
-			}
-			ipv4 = ip.String()
-		}
-		if ip.Is6() && r.Host == fmt.Sprintf("[%s]:%d", ip, ListenPort) {
-			return false
-		}
+	ipv4, ipv6 := s.selfNodeAddresses(r, st)
```

Comment:
Considered caching the "known IPs" for this node, but because it changes based on peer, opted to do the safer thing and look them up each time. It's a little more expensive than what we had before: now calling `Status` instead of `StatusWithoutPeers`, and calling `WhoIs` and extra time on the request.